consensus/src/liveness/leader_reputation.rs

```diff
@@ -240,14 +246,37 @@ impl NewBlockEventAggregation {
         history: &'a [NewBlockEvent],
```

Comment:
Is it possible to clone and re-sort the events here to be more robust instead of relying on the implicit order?